mubench.pipeline/data/detector_execution.py

```diff
@@ -69,13 +65,11 @@ def __init__(self, run_mode: DetectorMode, detector: Detector, version: ProjectV
         self.findings_filter = findings_filter
 
     def execute(self, compile_base_path: str, timeout: Optional[int], logger: Logger):
-        detector_invocation = ["java"] + self.detector.java_options + ["-jar", _quote(self.detector.jar_path)]
-        command = detector_invocation + self._get_detector_arguments(self.version.get_compile(compile_base_path))
-        command = " ".join(command)
+        detector_args = self._get_detector_arguments(self.version.get_compile(compile_base_path))
 
         start = time.time()
         try:
-            Shell.exec(command, logger=logger, timeout=timeout)
+            self.detector.runner_interface.execute(self.version, detector_args, timeout, logger)
```

Comment:
I was contemplating if the caller should check `if self.detector.runner_interface`. One way or another, we will probably have to do error handling explicitly when calling `execute`, since the interface doesn't have logging. That is why I avoided raising an exception in the interface.